### ISSUE TITLE
Certain syntaxes in ruby cause extraction errors

### ISSUE THREAD

**Issue by wonda-tea-coffee** [github/codeql#14279]
The following ruby code causes an extraction error when parsed.

```ruby
def foo(bar: nil) = raise NotImplementedError
```

The error message is as follows.

```
Error: Extraction failed in [filepath] with error A parse error occurred. Check the syntax of the file. If the file is invalid, correct the error or exclude the file from analysis.
```

This code is correct ruby syntax.
This problem was reproduced with CodeQL version 2.4.5.

**Comment by aibaars** [comment]
You're absolutely right, thanks for reporting. This is a problem with the underlying ruby parser that CodeQL uses. I raised an issue on tree-sitter-ruby: https://github.com/tree-sitter/tree-sitter-ruby/issues/242

The parse error is unlikely to cause problems with the CodeQL analysis because the parser should be able to recover and analyze the rest of the code normally. As a workaround you could use a parenthesized argument list in the one-line method `def`:

```ruby
def foo(bar: nil) = raise(NotImplementedError)
```

**Comment by wonda-tea-coffee** [comment]
Thank you very much!
I appreciate your prompt reply.
